### PR TITLE
Update project to support django 5.0

### PR DESCRIPTION
**Pull request by ldynia**
Update project to support django 5.0 and python 3.12

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -9,8 +9,9 @@ envlist =
     pylint
     # Python/Django combinations that are officially supported
     py3{6,7,8,9,10}-django32
-    py3{8,9,10,11}-django41
-    py3{8,9,10,11}-django42
+    py3{8,9,10,11,12}-django41
+    py3{8,9,10,11,12}-django42
+    py3{8,9,10,11,12}-django50
```
Comment (bittner) [inline tox.ini:14]: Django 5.0 requires Python >= 3.10 according to the [package metadata on PyPI](https://pypi.org/project/Django/).

After adjusting this line you also need to exclude Python < 3.10 for Django 5.0 in the [test matrix of GHA](https://github.com/behave/behave-django/blob/main/.github/workflows/test.yml#L32-L37).

**Review by bittner — CHANGES_REQUESTED** [review]

File: tox.ini (hunk 1 above)
Comment (ldynia) [reply tox.ini]: Thanks. I have made [relevant changes](https://github.com/behave/behave-django/pull/153/files).

**Review by bittner — CHANGES_REQUESTED** [review]
Great, looks good, thank you!

Could you update the test matrix also with the Python version, i.e. add Python 3.12, please? Django [5.0](https://pypi.org/project/Django/5.0.1/) and [4.2](https://pypi.org/project/Django/4.2.9/) officially support 3.12, Django [4.1](https://pypi.org/project/Django/4.1.13/) only goes up to 3.11, Django [3.2](https://pypi.org/project/Django/3.2.23/) only to 3.10.

**Comment by ldynia** [comment]
I have update the matrix please [review](https://github.com/behave/behave-django/pull/153/files)

**Review by bittner — CHANGES_REQUESTED** [review]
You're a hero! :1st_place_medal: 

Now, the tests for Python 3.12 fail for something that is not your fault! It would be awesome if you could fix that too, anyway.

The problem comes from [deprecated aliases](https://docs.python.org/3.11/library/unittest.html#deprecated-aliases) that were finally dropped in Python 3.12.

AFAICS, according to the tests, `assertEquals` needs to be replaced by `assertEqual` in the acceptance tests of the project.

```python
          File "tests/acceptance/steps/context-urlhelper.py", line 29, in is_baseurl_value
            context.test.assertEquals(context.result, context.base_url)
            ^^^^^^^^^^^^^^^^^^^^^^^^^
        AttributeError: 'BehaviorDrivenTestCase' object has no attribute 'assertEquals'. Did you mean: 'assertEqual'?
```

```python
          File "tests/acceptance/steps/context-urlhelper.py", line 34, in baseurl_plus_path
            context.test.assertEquals(context.result, context.base_url + url_path)
            ^^^^^^^^^^^^^^^^^^^^^^^^^
        AttributeError: 'BehaviorDrivenTestCase' object has no attribute 'assertEquals'. Did you mean: 'assertEqual'?
```

```python
          File "tests/acceptance/steps/context-urlhelper.py", line 54, in explicit_reverse
            context.test.assertEquals(context.result, context.get_url(path))
            ^^^^^^^^^^^^^^^^^^^^^^^^^
        AttributeError: 'BehaviorDrivenTestCase' object has no attribute 'assertEquals'. Did you mean: 'assertEqual'?
```

```python
          File "tests/acceptance/steps/context-urlhelper.py", line 60, in get_model_url
            context.test.assertEquals(context.result, context.get_url(path))
            ^^^^^^^^^^^^^^^^^^^^^^^^^
        AttributeError: 'BehaviorDrivenTestCase' object has no attribute 'assertEquals'. Did you mean: 'assertEqual'?
```

If you'd replace `assertEquals` in those places by `assertEqual` then the two failing test jobs for Python 3.12 will also pass.

**Comment by ldynia** [comment]
I have made the changes from `assertEquals` to `assertEqual`

**Review by bittner — APPROVED** [review]

**Comment by bittner** [comment]
Thank you for contributing! :100: :1st_place_medal:

**Comment by ldynia** [comment]
My motto is Live long and Pull Request :)

**Pull request merged**